### PR TITLE
support react16

### PR DESCRIPTION
**Pull request by brucewpaul**
this pr aims to resolve https://github.com/graphql/graphiql/issues/603 by supporting react16

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -57,8 +57,8 @@
   "peerDependencies": {
     "graphql": "^0.6.0 || ^0.7.0 || ^0.8.0-b || ^0.9.0 || ^0.10.0 || ^0.11.0",
     "prop-types": ">=15.5.0",
-    "react": "^15.6.0",
-    "react-dom": "^15.6.0"
+    "react": "^15.6.0 || ^16.0.0",
```
Comment (asiandrummer) [inline package.json]: Why not `^16.0.0` only?

**Comment by brucewpaul** [comment]
@asiandrummer updated!

**Comment by wincent** [comment]
> Why not `^16.0.0` only?

Because peer dependencies should be liberal. You don't want to force people to upgrade unless there is something in the new version that the project *needs*.

**Comment by brucewpaul** [comment]
@wincent @asiandrummer 

I agree that it should be more liberal and have updated it to reflect that

**Comment by jahed** [comment]
Is this PR going to get merged and release any time soon? Any blockers?

**Comment by ajhyndman** [comment]
I'd also love to see this go forward!

**Comment by brucewpaul** [comment]
@wincent @asiandrummer can this be merged/released?

File: package.json [hunk 2]
```diff
@@ -90,9 +90,9 @@
     "postcss-cli": "4.1.0",
     "prettier": "^1.4.4",
     "prop-types": "15.5.8",
-    "react": "15.5.4",
-    "react-dom": "15.5.4",
-    "react-test-renderer": "15.6.1",
+    "react": "15.5.4 || ^16.0.0",
+    "react-dom": "15.5.4 || ^16.0.0",
+    "react-test-renderer": "15.5.4 || ^16.0.0",
```
Comment (lencioni) [inline package.json]: I'd probably make the dev deps more explicit. Also, I think you may have changed the version here by accident.

**Comment by wincent** [comment]
Thanks @brucewpaul!

**Pull request merged**